### PR TITLE
feat: adds loginWithUsername option to auth config

### PR DESCRIPTION
**Pull request by JessRynkar**
## Description

Adds `loginWithUsername` option to auth config. When set to true, it will inject an `username` field into the collection config which replaces the `email` field in the UI. The `email` field is still required but not unique.

The `username` field can be extended by passing a field named `username` to your auth collection. Anything added to this field will be combined with the initial field.

- [X] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] Existing test suite passes locally with my changes

### CODE REVIEW

File: packages/graphql/src/resolvers/auth/forgotPassword.ts [hunk 1]
```diff
@@ -1,15 +1,18 @@
 import type { Collection } from 'payload'
 
 import { forgotPasswordOperation, isolateObjectProperty } from 'payload'
+import { use } from 'react'
```
Comment (JarrodMFlesch) [inline packages/graphql/src/resolvers/auth/forgotPassword.ts:4]: 😄

File: packages/graphql/src/resolvers/auth/forgotPassword.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 
 import type { Context } from '../types.js'
 
 function forgotPasswordResolver(collection: Collection): any {
   async function resolver(_, args, context: Context) {
+    console.log(args)
```
[the remark this answers is not in the document]
Comment (JessChowdhury) [reply packages/graphql/src/resolvers/auth/forgotPassword.ts]: F

File: packages/graphql/src/resolvers/auth/forgotPassword.ts (hunk 1 above)
Comment (JessChowdhury) [reply packages/graphql/src/resolvers/auth/forgotPassword.ts]: Sheesh

File: packages/payload/src/collections/config/sanitize.ts [hunk 3]
```diff
@@ -135,6 +136,12 @@ export const sanitizeCollection = async (
     }
 
     if (!sanitized.auth.disableLocalStrategy) {
+      const loginField = sanitized.auth.loginWithUsername ? 'username' : 'email'
+
+      if (loginField) {
+        authFields = authFields.concat(baseLoginField(loginField))
+      }
```
Comment (JarrodMFlesch) [inline packages/payload/src/collections/config/sanitize.ts:143]: This if statement is not needed, loginField will always be username or email. Was the intention to do something different?

Comment (JessChowdhury) [reply packages/payload/src/collections/config/sanitize.ts]: Nope not needed - removed!

**Pull request merged**